packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -25,18 +87,52 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
 	 * or else logging will not occur.
 	 */
 	private readonly baseLoggingClient: ApplicationInsights;
+	private readonly config: FluidAppInsightsLoggerConfig;
+	private readonly filters: TelemetryFilter[] = [];
 
-	public constructor(client: ApplicationInsights) {
+	public constructor(client: ApplicationInsights, config?: FluidAppInsightsLoggerConfig) {
 		this.baseLoggingClient = client;
+		this.config = config ?? {
+			filterConfig: {
+				mode: "exclusive",
+				filters: [],
+			},
+		};
+
+		if (config?.filterConfig.filters) {
+			this.filters = config.filterConfig.filters;
+		}
 	}
 
 	/**
-	 * Routes Fluid telemetry events to the trackEvent App Insights API
+	 * Routes Fluid telemetry events to the trackEvent App Insights API.
+	 * This method also uses the provided {@link FluidAppInsightsLoggerConfig} to
+	 * determine whether an event should be sent or not.
 	 */
 	public send(event: ITelemetryBaseEvent): void {
-		this.baseLoggingClient.trackEvent({
-			name: event.eventName,
-			properties: event,
-		});
+		// By default, "inclusive" filter mode means all events should not be sent by default
+		// and the opposite is true for "exclusive".
+		let shouldSendEvent = this.config.filterConfig.mode === "inclusive" ? false : true;
+		if (this.filters.length > 0 && this.doesEventMatchFilter(event)) {
+			// If the event does match a filter, in "inclusive" filter mode that means it should
+			// be sent (included). In "exclusive" mode the opposite is true.
+			shouldSendEvent = this.config.filterConfig.mode === "inclusive" ? true : false;
+		}
+
+		if (shouldSendEvent) {
```

Comment:
What do you think of encapsulating all this (and I'd say even the loop over filters) in a `shouldSendEvent()` function? So the send method can be just

```typescript
public send(event: ITelemetryBaseEvent): void {
	if (shouldSendevent(event)) {
		this.baseLoggingClient.trackEvent({
			name: event.eventName,
			properties: event,
		});
	}
}
```

Reply:
Yeah that is cleaner separation of concerns. I have updated to this.